rsocket-core/src/main/java/io/rsocket/transport/TransportHeaderAware.java

```diff
@@ -0,0 +1,12 @@
+package io.rsocket.transport;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Extension interface to support Transports with headers at the transport layer,
+ * e.g. Websockets, Http2.
+ */
+public interface TransportHeaderAware {
```

Comment:
wondering whether this should return a copy of the Transport with these headers applied?